tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/typeof/TypeOfRegressionTest.java

```diff
@@ -73,4 +73,15 @@ public void testLiteralType() {
         assertEquals(type.get().kind(), SymbolKind.TYPE);
         assertEquals(type.get().typeKind(), TypeDescKind.INT);
     }
+
+    @Test
+    public void testResourcePathAccess() {
+        model =SemanticAPITestUtils.getDefaultModulesSemanticModel(
+                "test-src/regression-tests/typeof_resource_path_action.bal");
+
+        Optional<TypeSymbol> type = model.typeOf(LineRange.from("typeof_resource_path_action.bal",
+                LinePosition.from(20, 23), LinePosition.from(20, 30)));
+
```

Comment:
```suggestion
```

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/40026/commits/351f502a52d815ecbb7713a6a94d19d3ecf5a4a2.